main.cpp

```diff
@@ -19,13 +18,31 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+	CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) { }
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+	virtual void run(const MatchFinder::MatchResult& Result) {
+	const auto *CastExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");

```

Comment:
What if `CastExpr` is a null pointer? How to protect the program from crashing in such cases?

Reply:
I can add nullptr guard
```
if (!CastExpr) {
    return;
}
```